llarp/router/router.cpp

```diff
@@ -97,6 +97,105 @@ namespace llarp
     }
   }
 
+  util::StatusObject
+  Router::ExtractSummaryStatus() const
+  {
+    if (_running)
+    {
+      auto services = _hiddenServiceContext.ExtractStatus();
+      auto link_types = _linkManager.ExtractStatus();
+
+      int tx_rate = 0;
+      int rx_rate = 0;
+      int peers = 0;
+      for (const auto& links : link_types)
+      {
+        if (links == nlohmann::json::array())
+          continue;
+        for (const auto& link : links)
+        {
+          if (links == nlohmann::json::array())
+            continue;
+          const auto& established_peers = link["sessions"]["established"];
+          for (const auto& peer : established_peers)
+          {
+            tx_rate += peer["tx"].get<int>();
+            rx_rate += peer["rx"].get<int>();
+            peers++;
+          }
+        }
+      }
+
+      std::string loki_address = services["default"]["identity"];
+
+      std::string exit_node;
+      auto exit_map = services["default"]["exitMap"];
+      if (!exit_map.is_null())
+      {
+        // exitMap should be of length 1 only, but it's an object with keys an IP (not as string)
```

Comment:
this assumption of there being only one entry is not correct, we use a map because there are more than one.